File: DSA/Anisha_Parveen/Day_12_Ques_2.cpp
#include <bits/stdc++.h>
using namespace std;

int singleDigitNumber(int number)
{
    if (number < 10)
        return number;

    int sum = 0;

    while (number > 0)
    {
        sum += number % 10;
        number /= 10;
    }

    return singleDigitNumber(sum);
}

int main()
{
    int number;
    cout << "Enter a number: ";
    cin >> number;

    int ans = singleDigitNumber(number);
    cout << "The single digit result is: " << ans << endl;
}

File: DSA/Asmi_Garg/day12_ques2.java
import java.math.BigInteger;
import java.util.Scanner;

public class DigitSum {

    public static int sumDigits(BigInteger number) {
        if (number.compareTo(BigInteger.TEN) < 0) {
            return number.intValue();
        } else {
            BigInteger sum = BigInteger.ZERO;
            while (number.compareTo(BigInteger.ZERO) > 0) {
                sum = sum.add(number.mod(BigInteger.TEN));
                number = number.divide(BigInteger.TEN);
            }
            return sumDigits(sum);
        }
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        System.out.print("Enter a large integer: ");
        BigInteger number = scanner.nextBigInteger();

        int result = sumDigits(number);
        System.out.println("The single digit result is: " + result);

        scanner.close();
    }
}
